go/pkg/libproxy/multiplexed.go

```diff
@@ -14,29 +14,28 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-const (
-	maxBufferSize = 65536
-)
+const defaultWindowSize = 65536
 
 type windowState struct {
 	current uint64
 	allowed uint64
+	max     uint64
 }
 
 func (w *windowState) String() string {
-	return fmt.Sprintf("current %d, allowed %d", w.current, w.allowed)
+	return fmt.Sprintf("current %d, allowed %d, max %d", w.current, w.allowed, w.max)
 }
 
 func (w *windowState) size() int {
 	return int(w.allowed - w.current)
 }
 
 func (w *windowState) isAlmostClosed() bool {
-	return w.size() < maxBufferSize/2
+	return w.size() < int(w.max/2)
```

Comment:
What if w.max is set to an odd value ?

Reply:
Normally it doesn't send window updates (unlike TCP), it only sends one when `isAlmostClosed`. If `max` was odd, then `max / 2` would round down, and it would send a window update 1 byte earlier. To check nothing bad happens, I've pushed a patch to set the max to `defaultWindowSize-1` in one of the test cases.